bsb/connectivity/point_cloud/cloud_cloud_intersection.py

```diff
@@ -5,7 +5,7 @@
 from bsb import config
 from bsb.connectivity.strategy import Hemitype
 from .geometric_shapes import ShapeCompositionDependencyNode
-
+from bsb.trees import BoxRTree
```

Comment:
```suggestion
from bsb.trees import BoxTree
```

`BoxRTree` should be private, I'll rectify that on `main`

Reply:
Here is an rough idea of implementation for `BoxTree`:
```python
pre_pos = pre_ps.load_positions()
post_pos = post_ps.load_positions()
pre_cloud_cache = self.presynaptic.shape_compositions
post_cloud_cache = self.postsynaptic.shape_compositions
tree = BoxTree([np.concatenate([cloud._mbb_min + pre_coord, 
                                cloud._mbb_max + pre_coord]) for pre_coord, cloud in zip(pre_pos, pre_cloud_cache)])
inside_mbbox = tree.query([np.concatenate([cloud._mbb_min + post_coord, 
                                           cloud._mbb_max + post_coord]) for post_coord, cloud in zip(post_pos, post_cloud_cache)])
```
While I understand BoxTree can be useful to reduce the time it takes to filter `ShapesComposition`s that overlap, I wonder if it is worth it since most neurons can be represented with a single `ShapesComposition`, and the overlapping of their bounding box is quite efficient. 
@Helveg, @alessiomarta, do you see any scenario where we would need multiple `ShapesComposition` objects to represent the pre or postsynaptic neurons? Otherwise we can force the user to use a single `ShapesComposition` per neuron. Then, in this configuration, would it be still interesting to use `BoxTree`?